spec/rubocop/cli/options_spec.rb

```diff
@@ -1235,6 +1235,70 @@ def full_description_of_cop(cop)
     end
   end
 
+  describe '--json-schema' do
+    let(:stdout) { $stdout.string }
+    let(:cmd) { cli.run(['--json-schema']) }
+    let(:parsed_schema) do
+      cmd
+      JSON.parse(stdout)
+    end
+    let(:properties) { parsed_schema['properties'] }
+
+    it 'runs without errors' do
+      # run directly rather than through CLI command so we get errors
+      env = RuboCop::CLI::Environment.new(nil, nil, nil)
+      cmd = RuboCop::CLI::Command::JSONSchema.new(env)
+      expect { cmd.run }.not_to raise_error
+    end
+
+    it 'generates valid json' do
+      expect { parsed_schema }.not_to raise_error
+    end
+
+    it 'has a top-level properties key' do
+      expect(parsed_schema.key?('properties')).to be(true)
+    end
+
+    it 'knows about top-level options' do
+      expect(properties.key?('require')).to be(true)
+      expect(properties.key?('inherit_from')).to be(true)
+      expect(properties.key?('AllCops')).to be(true)
+    end
+
+    # without resolving json pointers it's a bit tricky to test specific
+    # attributes. skip it for now to avoid bloating this file
+    it 'knows about departments' do
+      expect(properties.key?('Layout')).to be(true)
+    end
+
+    it 'knows about cops' do
+      expect(properties.key?('Layout/MultilineAssignmentLayout')).to be(true)
+    end
+
+    it 'knows about lots of cops' do
+      cop_props = properties.keys.select { |k| k.include?('/') }
+      expect(cop_props.length).to be > 100
+    end
+
+    it 'knows about departments in other cities' do
```

Comment:
I don't think we've used "city" before as part of RuboCop lingo. `InternalAffairs` is technically a _custom department,_ I believe. Its cops are loaded because of a `require` in RuboCop's root `.rubocop.yml`.

Reply:
yeah probably too cute, I'll change it to "extension", which I think is the term used elsewhere.

I wanted to use the `Performance` rather than `InternalAffairs` as the test-case for this but had a little bit of trouble.